sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusMessageBatch.java

```diff
@@ -26,13 +25,12 @@
  */
 public final class ServiceBusMessageBatch {
     private final ClientLogger logger = new ClientLogger(ServiceBusMessageBatch.class);
-    private final Object lock = new Object();
     private final int maxMessageSize;
     private final ErrorContextProvider contextProvider;
     private final MessageSerializer serializer;
     private final List<ServiceBusMessage> serviceBusMessageList;
     private final byte[] eventBytes;
-    private final AtomicInteger sizeInBytes;
```

Comment:
Why remove `AtomicInteger`? I assume this class still need to be thread-safe?

Reply:
I see. This appear to be a data object, which indeed might not need to be thread-safe.

Another question is why `LinkedList` instead of `ArrayList` if not need to be thread-safe. `ArrayList` is more cache-friendly.

Reply:
I use `LinkedList` because originally the code use `LinkedList`.  I see in ManagementChannel.java also use `LinkedList` to save message. https://github.com/Azure/azure-sdk-for-java/blob/9b387878ab7f325a9ba32c2bc8162c664e064731/sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ManagementChannel.java#L320 

But here `serviceBusMessageList` is just keep adding message one by one and then do the iteration. I think `ArrayList` may have better performance. I will check that. Thanks for the suggestion.